File: software/examples/php/ExampleSimple.php
<?php

require_once('Tinkerforge/IPConnection.php');
require_once('Tinkerforge/BrickletCompass.php');

use Tinkerforge\IPConnection;
use Tinkerforge\BrickletCompass;

const HOST = 'localhost';
const PORT = 4223;
const UID = 'XYZ'; // Change XYZ to the UID of your Compass Bricklet

$ipcon = new IPConnection(); // Create IP connection
$c = new BrickletCompass(UID, $ipcon); // Create device object

$ipcon->connect(HOST, PORT); // Connect to brickd
// Don't use device before ipcon is connected

// Get current heading
$heading = $c->getHeading();
echo "Heading: " . $heading/10.0 . " °\n";

echo "Press key to exit\n";
fgetc(fopen('php://stdin', 'r'));
$ipcon->disconnect();

?>

File: software/examples/python/example_simple.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

HOST = "localhost"
PORT = 4223
UID = "XYZ" # Change XYZ to the UID of your Compass Bricklet

from tinkerforge.ip_connection import IPConnection
from tinkerforge.bricklet_compass import BrickletCompass

if __name__ == "__main__":
    ipcon = IPConnection() # Create IP connection
    c = BrickletCompass(UID, ipcon) # Create device object

    ipcon.connect(HOST, PORT) # Connect to brickd
    # Don't use device before ipcon is connected

    # Get current heading
    heading = c.get_heading()
    print("Heading: " + str(heading/10.0) + " °")

    raw_input("Press key to exit\n") # Use input() in Python 3
    ipcon.disconnect()
